### PR TITLE
Add all students view with basic export functionality

### PR DESCRIPTION
**Pull request by Javi**
Addresses #226, requires https://github.com/mumuki/mumuki-classroom-api/pull/167

### CODE REVIEW

**Comment by flbulgarelli** [comment]
@luchotc please review

File: src/scripts/controllers/students-controller.js [hunk 1]
```diff
@@ -24,29 +29,31 @@ angular
     $scope.canDetach = Permissions.isJanitor();
     $scope.canAddStudent = Permissions.isJanitor();
 
-    $scope.setCount(students.total);
+    //TODO: Uncouple from course controller
+    $scope.setCount && $scope.setCount(students.total);
```
Comment (luchotc) [inline src/scripts/controllers/students-controller.js]: I agree with this but i don't think it's worth extracting another controller yet. We could take advantage of the if in line 18, and only call setCount if stateParams does include courses. What do you think?

File: src/scripts/controllers/students-controller.js [hunk 2]
```diff
@@ -77,6 +84,13 @@ angular
 
     $scope.transfer = (student) => {
       Modal.transfer(student, () => $state.reload());
-    }
+    };
+
+    $scope.exportStudents = () => {
```
Comment (luchotc) [inline src/scripts/controllers/students-controller.js]: I showed this to @aguspina and he asked me if we could keep the same reports of /courses/students or at least the global report to be a CSV.

File: src/scripts/services/breadcrumb-service.js [hunk 3]
```diff
@@ -5,74 +5,75 @@ angular
...
       return this._students;
-    }
+    };
 
     this._course = null;
     this._guide = null;
     this._student = null;
     this._list = [
       { name: () => 'notifications', state: 'classroom.notifications' },
+      { name: () => 'students', state: 'classroom.students.list' },
```
Comment (luchotc) [inline src/scripts/services/breadcrumb-service.js]: This makes the breadcrumb to be /courses/students. I would prefer just /students, but if it is a lot of work it's all right!

File: src/views/select.jade [hunk 4]
```diff
@@ -17,5 +19,8 @@
     label(ng-if='!(list | orderBy: sortCriteria() | filter:listSearch:searchComparator).length')
       | {{ noItemsToShow | translate }}
 
+  button.btn.btn-primary(ng-if='isJanitor()', ui-sref='classroom.students.list')
```
Comment (luchotc) [inline src/views/select.jade]: @aguspina also asked me to change the text of this button to "Estadisticas generales".

**Comment by flbulgarelli** [comment]
@luchotc or @Javi could you please provide some screenshots?

**Comment by julian-berbel** [comment]
> @luchotc or @Javi could you please provide some screenshots?

image
image

File: src/scripts/controllers/students-controller.js [hunk 5]
```diff
@@ -77,6 +84,9 @@ angular
 
     $scope.transfer = (student) => {
       Modal.transfer(student, () => $state.reload());
-    }
+    };
 
+    $scope.reportUrl = () => {
+      return `${Domain.classroomApiURL()}/students/report`;
```
Comment (julian-berbel) [inline src/scripts/controllers/students-controller.js]: Requires mumuki/mumuki-classroom-api#172.

File: src/scripts/controllers/students-controller.js (hunk 1 above)
Comment (julian-berbel) [reply src/scripts/controllers/students-controller.js]: 👍

**Comment by flbulgarelli** [comment]
I feel weird that button there. I'd expect a tab named "students" under `/students` route.

**Comment by luchotc** [comment]
@flbulgarelli Yep, i quite feel the same. I'm not sure how complex it is to change it, and if it's worth

**Comment by aguspina** [comment]
Maybe another item named "All students" in the profile dropdown ul?

**Comment by flbulgarelli** [comment]
Ok, lets keep the button. I will add a separate issue for that.

**Pull request merged**